### PR TITLE
Docker login 

### LINKED ISSUE #506: Support private images on Docker Hub
Something along the lines of:

```
fn docker login ...
```

See (or copy) https://github.com/iron-io/ironcli

Runner should already support this, we'll have to pass the info into it though.

### PR DESCRIPTION
**Pull request by BupycHuk**
closes #506

### CODE REVIEW

File: fn/login.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		Context: context.Background(),
+	}
+
+	res, err := l.client.Docker.PostDockerLogin(params)
+	fmt.Fprintf(os.Stdout, "res := %v", res)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "err: %v", err)
+		return fmt.Errorf("unexpected error: %v", err)
+	}
+	fmt.Println(`Added docker repo credentials`)
```
Comment (treeder) [inline fn/login.go]: Why backticks?

**Review by treeder — CHANGES_REQUESTED** [review]
I wonder about the security implications of this. It may be that this features requires an authentication module or something?

**Comment by BupycHuk** [comment]
@treeder 
I'm going to encrypt docker credentials using AES or RSA.
Do you think this is enough?

File: api/server/server.go [hunk 2]
```diff
@@ -69,11 +71,16 @@ func NewFromEnv(ctx context.Context) *Server {
 
 	apiURL := viper.GetString(EnvAPIURL)
 
-	return New(ctx, ds, mq, apiURL)
+	auth := auth.DockerAuth{
+		Datastore: ds,
+		Key:       []byte("AES256Key-32Characters1234567890"),
```
Comment (treeder) [inline api/server/server.go]: User should pass this in through env var perhaps?

**Comment by treeder** [comment]
Maybe we start here with global docker credentials, then once we add some authentication middleware, those could also have per user credentials.

**Comment by treeder** [comment]
@BupycHuk has some conflicts here.

**Comment by BupycHuk** [comment]
@treeder i have fixed conflicts and add new tests, could you take a look, pls

**Pull request closed without merging**